tests/pyccel/test_pyccel.py

```diff
@@ -849,6 +849,44 @@ def test_module_init( language ):
     compare_pyth_fort_output(pyth_out, lang_out, str, language)
 
 #------------------------------------------------------------------------------
+def get_lang_exit_value(abs_path, language, cwd=None):
+    abs_path = get_exe(abs_path, language)
+    if language == "python":
+        if cwd is None:
+            p = subprocess.Popen([sys.executable , "%s" % abs_path], stdout=subprocess.PIPE, universal_newlines=True)
+        else:
+            p = subprocess.Popen([sys.executable , "%s" % abs_path], stdout=subprocess.PIPE, universal_newlines=True, cwd=cwd)
+    else:
+        p = subprocess.Popen(["%s" % abs_path], stdout=subprocess.PIPE, universal_newlines=True)
+    p.communicate()
+    return p.returncode
+
+def pyccel_test_program_exit(language, test_file):
+    test_dir = os.path.dirname(test_file)
+    test_file = get_abs_path(os.path.normpath(test_file))
+
+    output_dir   = os.path.join(get_abs_path(test_dir),'__pyccel__')
+    output_test_file = os.path.join(output_dir, os.path.basename(test_file))
+
+    cwd = get_abs_path(test_dir)
+
+    if not language:
+        language = "fortran"
+    pyccel_commands = " --language="+language
+    pyccel_commands += " --output="+ output_dir
+
+    compile_pyccel(cwd, test_file, pyccel_commands)
+    lang_out = get_lang_exit_value(output_test_file, language)
+    pyth_out = get_lang_exit_value(test_file, "python")
+    assert (not lang_out and not pyth_out) or (lang_out and pyth_out)
+
+
+def test_assert(language ):
+    pyccel_test_program_exit(language, "scripts/asserts/valid_assert.py")
+    pyccel_test_program_exit(language, "scripts/asserts/unvalid_assert1.py")
+    pyccel_test_program_exit(language, "scripts/asserts/unvalid_assert2.py")
+    pyccel_test_program_exit(language, "scripts/asserts/unvalid_assert3.py")
```

Comment:
I suggest using `pytest.mark.parametrize` to make this into 3 tests. See `test_hope_benchmarks` for an example

Reply:
Modified in 972b8a5d76c19b749bba3bd48cf7e75bc7f9afb3.
https://github.com/pyccel/pyccel/blob/972b8a5d76c19b749bba3bd48cf7e75bc7f9afb3/tests/pyccel/test_pyccel.py#L883-L890
is this better ?

Reply:
Looks good. Is there any need for `test_assert` and `pyccel_test_program_exit` to be 2 files now? Do you think `pyccel_test_program_exit` could be useful for testing things other than asserts?

Reply:
I combined `test_assert` and `pyccel_test_program_exit`  in b62266bfdc397d3fb1d330066c543c61eacb1a5a.